src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const updatedState = [];
+  let tempObject = { ...state };
+
+  for (const action of actions) {
+    switch (true) {
+      case action.type === 'addProperties':
+        add(action.extraData);
+        break;
+      case action.type === 'removeProperties':
+        remove(action.keysToRemove);
+        break;
+      case action.type === 'clear':
+        clear();
+        break;
+    }
```

Comment:
Consider using a regular switch statement with action.type as the expression. The current usage of switch with `true` and multiple case checks is unconventional and may lead to confusion. It is also not as optimized as using the value to be checked directly in the switch statement.